---
title: "BINFF401"
author: "Thomas De Deyn"
output: html_document
date: "`r Sys.Date()`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo = FALSE}
library(tidyverse)
library(corrplot)
library(grid)
library(gridExtra)
library(caret)
library(backports)
library(DESeq2)
library(EBImage)

```

# BINF-F401 project (R Markdown)

The task assigned was to study the RNA expression of a set of patients. First, let's set the parameters used throughout the scripts.

```{r}
  p_value = .05
  discrete = 20
```

## Morphology vs Gene Expression

### 1. Report:

-   The number of significant down-regulated and up-regulated genes associated with each morphological cluster
-   The 10 most significant up-regulated genes

```{r, echo = FALSE}
morphological_counts = read_tsv("D:/Polytech/MA2 2023-2024/Q2/Genomics/Project_ThyroidGland/Data_ThyroidGland/OG/morphological_counts_lunit_dino.tsv")
coldata = morphological_counts
RNA_read = read_tsv("D:/Polytech/MA2 2023-2024/Q2/Genomics/Project_ThyroidGland/Data_ThyroidGland/OG/RNA_read_counts.tsv")

clinical = read_tsv("D:/Polytech/MA2 2023-2024/Q2/Genomics/Project_ThyroidGland/Data_ThyroidGland/OG/clinical_data.tsv")
coldata = coldata %>% left_join(by = "SMPLID", y = clinical)

```

### -----------------------------------------------------------------------------------------------------

### Step 1: Filtering transcripts

Objective is to reduce the amount of transcript. Two things could be sought for when removing transcripts: A. Transcripts with low expression B. Transcripts with low variability (estimated using the the median average deviation)

```{r, echo = FALSE}

rownames(RNA_read) = RNA_read$Name 
df = RNA_read %>% select_if(is.numeric)
rownames(df) = RNA_read$Name 

otsu_modified = function(array){
  # Find the 5% largest values, and remove them (outliers)
  array <- sort(array)
  array <- array[1:(length(array) - 0.05 * length(array))]
  
  # Discretise data for otsu method
  
  
  # Compute the threshold
  return (otsu(log10(as.matrix(array, nrow = 2) + 1), 
            range = c(min(array, na.rm = TRUE), max(array, na.rm = TRUE)),
                      levels = discrete) )
  
}

```

```{r, echo = FALSE}

maxs = apply(df,1, function(var) max(var, na.rm = TRUE))

hist(maxs)
hist(log10(maxs + 1))


```

```{r, echo = FALSE}

median = apply(df,1, function(var) median(var, na.rm = TRUE))
thr_median = otsu_modified(median)

hist(log10(median + 1), breaks = discrete)
abline(v = log10(thr_median + 1), col = "red")
```

```{r, echo = FALSE}

med_avg = apply(df,1, function(var) mad(var, na.rm = TRUE))
thr_avg = otsu_modified(med_avg)

hist(log10(med_avg + 1), breaks = discrete) 
abline(v = log10(thr_avg + 1), col = "red")
```

```{r, echo = FALSE}

high_mad = med_avg > thr_avg
high_expr = median > thr_median 

```

Filtering steps:

```{r, echo = FALSE}

cond = (high_mad & high_expr)
print(sum(cond))
cts = df[cond, ]
rownames(cts) = rownames(df)[cond]

rownames(coldata) = coldata$SMPLID
#Checking if coldata is fine: 
all(colnames(cts) %in% rownames(coldata))
#Checking if same order: 
all(colnames(cts) == rownames(coldata))

```

## Perfomring DESEq2 analysis

```{r, echo = FALSE}

#varname = c("Mophological.cluster.G4_0", "Mophological.cluster.G4_1" )
varname = morphological_counts %>% select(-SMPLID) %>% colnames()

covar = c("Mophological.cluster.G4_23")
#G4_23 --> Muscle
#G4_26 --> Peu de cellules, Ã  utiliser? 

eq_cov = paste0(covar, collapse = " + ")
res = c()



for(var in varname){
  print(var)
  start = paste("~", var )
  dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = as.formula(paste(start, eq_cov, sep =" + ")))
  
  dds <- DESeq(dds)
  res[var] = list(results(dds, alpha = p_value))
}



```
12:32 (start) 
13:26 (end)
### Select variable of interests

```{r, echo = FALSE}
#saveRDS(res, file ="D:/Polytech/MA2 2023-2024/Q2/Genomics/Project_ThyroidGland/Data_ThyroidGland/OG/res_AGE_MC23")

top10iN32 = c()

for(var in varname){
  print(var)
  
  # Select significant regulations
  signif = res[[var]]
  #Some values are NA: 
  signif$padj[is.na(signif$padj)] = 0.555555555555
  signif = signif[signif$padj < p_value,]
  
  #Select up-regulations
  upreg = signif
  #Some values are NA: 
  upreg$pvalue[is.na(upreg$log2FoldChange)] = 0
  upreg = upreg[upreg$log2FoldChange > 0,]
  
  hist(upreg$log2FoldChange, main = paste("log2FoldChange of", sep = " ", var))
  
  # Order the dataframe, objective is to have the max values at the top
  max(upreg$log2FoldChange)
  upreg = upreg[order(-upreg$log2FoldChange),]
  
  print("The ten most significant up-regulated genes are:")
  
  tmp = upreg[1:10,]
  tmp$cluster = var
  top10iN32 = rbind(top10iN32, tmp)
  
  summary(res[[var]])
  summary(upreg)
  
  plotMA(res[[var]], main = paste("plotMA of", sep = " ", var))
  plotMA(upreg, , main = paste("plotMA of up-regulated in", sep = " ", var))
}

```

### Selection of the 10 most up-regulated genes

```{r, echo = FALSE}

top10iN32 = top10iN32[order(-top10iN32$log2FoldChange),]
top10iN32$gene = rownames(top10iN32)

top10 = c()
top10 = rbind(top10, top10iN32[1, ])

for (r in 2:nrow(top10iN32)){
 if(! (top10iN32[r, ]$gene %in% top10$gene)){
   top10 = rbind(top10, top10iN32[r, ])
 }
}
```